nixos/modules/system/boot/networkd.nix

```diff
@@ -2995,13 +2998,18 @@ let
       systemd.sockets.systemd-networkd = {
         wantedBy = [ "initrd.target" ];
       };
+      systemd.services.systemd-resolved = {
+        wantedBy = ["initrd.target"];
```

Comment:
The upstream unit file has `WantedBy=sysinit.target`, so we should probably do the same. Same goes for the stage 2 unit, really.